compiler/src/typed/env.re

```diff
@@ -665,6 +665,7 @@ let get_components = c =>
 type compilation_mode =
   | Normal
   | Runtime
+  | ManagedRuntime
```

Comment:
I added this mode as a sort of final layer between when GC is available but you don't have `Pervasives` yet. I'm not 100% thrilled with it, since it's essentially just a "don't include Pervasives" mode. We've talked about potentially being able to enable/disable compiler flags via directives in a file, but we never really landed on anything concrete.

Reply:
Can we add some doc comments to the different `compilation_mode` entries (either here or in `env.rei`)? I feel like it's becoming fuzzy (to me) what each means...like, I feel like it's going to be easy to forget what the difference between `Runtime` and `ManagedRuntime` is once we context switch away from these PRs you've recently opened.

Reply:
Yeah, I can copy the descriptions over.